clippy_lints/src/transmute/transmute_ref_to_ref.rs

```diff
@@ -19,6 +18,8 @@ pub(super) fn check<'tcx>(
 ) -> bool {
     let mut triggered = false;
 
+    let mut applicability = Applicability::MachineApplicable;
+    let arg_sugg = sugg::Sugg::hir_with_context(cx, arg, e.span.ctxt(), "..", &mut applicability);
```

Comment:
The snippet should not be taken if the lint is not triggered. By using a closure, `arg_sugg()` can be used instead of `arg_sugg` later.

```suggestion
    let arg_sugg = || sugg::Sugg::hir_with_context(cx, arg, e.span.ctxt(), "..", &mut Applicability::Unspecified);
```